generate_analytical_dataset.py

```diff
@@ -0,0 +1,274 @@
+    for time_stamp_fmt in valid_timestamps:
+        try:
+            return datetime.strptime(time_stamp_as_string, time_stamp_fmt)
+        except Exception:
+            print(f'timestampAsString did not match valid format {time_stamp_fmt}')
+
+    # TODO Think about the below exception later on
+    raise Exception(f'Unparseable date found: {time_stamp_as_string} , did not match any supported date formats {valid_timestamps}')
+
+def sanitize(z):
+    decrypted = z['decrypted']
+    db_name = z['db_name']
+    collection_name = z['collection_name']
+    if ((db_name == "penalties-and-deductions" and collection_name == "sanction")
+            or (db_name == "core" and collection_name == "healthAndDisabilityDeclaration")
+            or (db_name == "accepted-data" and collection_name == "healthAndDisabilityCircumstances")):
+        decrypted = re.sub(r'(?<!\\)\\[r|n]', '', decrypted)
+    return decrypted.replace("$", "d_").replace("\u0000", "").replace("_archivedDateTime", "_removedDateTime").replace("_archived", "_removed")
+
+def decrypt(y):
+    key = y['plain_text_key']
+    iv = y['iv']
+    ciphertext = y['dbObject']
+    iv_int = int(binascii.hexlify(base64.b64decode(iv)), 16)
+    ctr = Counter.new(AES.block_size * 8, initial_value=iv_int)
+    aes = AES.new(base64.b64decode(key), AES.MODE_CTR, counter=ctr)
+    y['decrypted'] = aes.decrypt(base64.b64decode(ciphertext))
+    return y
+
+def get_plaintext_key_calling_dks(r,keys_map):
+    cek = r['encryptedKey']
+    kek = r['keyEncryptionkeyId']
+    if keys_map.get(cek):
+        key = keys_map[cek]
+        print("Found the key in cache")
+        r['plain_text_key'] = key
+    else:
+        print("Didn't find the key in cache so calling dks")
+        key = call_dks(cek, kek)
+        keys_map[cek] = key
+        r['plain_text_key'] = key
+    return r
+
+def call_dks(cek, kek):
+    url = "${url}"
+    params = {"keyId": kek}
+    result = requests.post(
+        url,
+        params=params,
+        data=cek,
+        cert=("private_key.crt", "private_key.key"),
+        verify="analytical_ca.pem",
+    )
+    content = result.json()
+    return content["plaintextDataKey"]
+
+def getTuple(json_string):
+    record = json.loads(json_string)
+    encryption = record["message"]["encryption"]
+    dbObject = record["message"]["dbObject"]
+    encryptedKey = encryption["encryptedEncryptionKey"]
+    keyEncryptionkeyId = encryption["keyEncryptionKeyId"]
+    iv = encryption["initialisationVector"]
+    # TODO  exception handling , what if the  below fields don't exist in the json
+    db_name = record['message']['db']
+    collection_name = record['message']['collection']
+    return {
+        "encryptedKey": encryptedKey,
+        "keyEncryptionkeyId":keyEncryptionkeyId,
+        "iv":iv,
+        "dbObject":dbObject,
+        "db_name":db_name,
+        "collection_name":collection_name
+    }
+
+def getPrinted(values):
+    for x in values:
+        print(x)
```

Comment:
we should follow the syntax style like in other methods. get_printed(values)

also what are we printing here? should we be printing these values out

Reply:
i dont think you are using this method anywhere

Reply:
Thanks  for spotting it Levan . removed it now